src/dodal/devices/common_dcm.py

```diff
@@ -0,0 +1,76 @@
+from typing import Generic, TypeVar
+
+from ophyd_async.core import (
+    StandardReadable,
+)
+from ophyd_async.epics.core import epics_signal_r
+from ophyd_async.epics.motor import Motor
+
+
+class StationaryCrystal(StandardReadable):
+    def __init__(self, prefix):
+        super().__init__(prefix)
+
+
+class RollCrystal(StationaryCrystal):
+    def __init__(self, prefix):
+        with self.add_children_as_readables():
+            self.roll_in_mrad = Motor(prefix + "ROLL")
+        super().__init__(prefix)
+
+
+class PitchAndRollCrystal(StationaryCrystal):
+    def __init__(self, prefix):
+        with self.add_children_as_readables():
+            self.pitch_in_mrad = Motor(prefix + "PITCH")
+            self.roll_in_mrad = Motor(prefix + "ROLL")
+        super().__init__(prefix)
+
+
+Xtal_1 = TypeVar("Xtal_1", bound=StationaryCrystal)
+Xtal_2 = TypeVar("Xtal_2", bound=StationaryCrystal)
+
+
+class BaseDCM(StandardReadable, Generic[Xtal_1, Xtal_2]):
+    """
+    Common device for the double crystal monochromator (DCM), used to select the energy of the beam.
+
+    Features common across all DCM's should include virtual motors to set energy/wavelength and contain two crystals,
+    each of which can be movable. Some DCM's contain crystals with roll motors, and some contain crystals with roll and pitch motors.
+    This base device accounts for all combinations of this.
+
+    This device should act as a parent for beamline-specific DCM's, in which any other missing signals can be added.
+
+    Bluesky plans using DCM's should be typed to specify which types of crystals are required
```

Comment:
Is there any example for this? Typing those plans may not be super easy on the first go, so it might be nice to either link or provide an example in the docstring...